bundles/binding/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/MyqGenericBindingProvider.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.myq.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.openhab.binding.myq.MyqBindingProvider;
+import org.openhab.binding.myq.internal.MyqBindingConfig.ITEMTYPE;
+import org.openhab.core.items.Item;
+import org.openhab.core.library.items.RollershutterItem;
+import org.openhab.core.library.items.SwitchItem;
+import org.openhab.core.library.items.ContactItem;
+import org.openhab.core.library.items.StringItem;
+import org.openhab.model.item.binding.AbstractGenericBindingProvider;
+import org.openhab.model.item.binding.BindingConfigParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is responsible for parsing the binding configuration.
+ * 
+ * @author Scott Hanson
+ * @since 1.8.0
+ */
+public class MyqGenericBindingProvider extends AbstractGenericBindingProvider
+		implements MyqBindingProvider {
+	static final Logger logger = LoggerFactory
+			.getLogger(MyqGenericBindingProvider.class);
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public String getBindingType() {
+		return "myq";
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	public void validateItemType(Item item, String bindingConfig)
+			throws BindingConfigParseException {
+		if (!(item instanceof SwitchItem || item instanceof RollershutterItem
+				|| item instanceof ContactItem || item instanceof StringItem)) {
+			throw new BindingConfigParseException(
+					"item '"
+							+ item.getName()
+							+ "' is of type '"
+							+ item.getClass().getSimpleName()
+							+ "', only SwitchItems, RollershutterItem, ContactItem or StringItem are allowed "
+							+ "- please check your *.items configuration");
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void processBindingConfiguration(String context, Item item,
+			String bindingConfig) throws BindingConfigParseException {
+		super.processBindingConfiguration(context, item, bindingConfig);
+		MyqBindingConfig config = parseBindingConfig(item, bindingConfig);
+
+		// parse bindingconfig here ...
+		addBindingConfig(item, config);
+	}
+
+	/**
+	 * Parse item type to see what the action is used for
+	 */
+	private MyqBindingConfig parseBindingConfig(Item item, String bindingConfig)
+			throws BindingConfigParseException {
+		final MyqBindingConfig config = new MyqBindingConfig();
+
+		if (item instanceof SwitchItem) {
+			config.type = ITEMTYPE.Switch;
+		} else if (item instanceof RollershutterItem) {
+			config.type = ITEMTYPE.Rollershutter;
+		} else if (item instanceof ContactItem) {
+			config.type = ITEMTYPE.ContactStatus;
+		} else if (item instanceof StringItem) {
+			config.type = ITEMTYPE.StringStatus;
+		}
```

Comment:
As mentioned before, instead of `ITEMTYPE`, consider
```
config.acceptedDataTypes = new ArrayList<Class<? extends State>>(item.getAcceptedDataTypes());
```
and stay unaware of the different item classes altogether.  Just post an update by looping the array and finding a type you support.